prometheus-client/src/Prometheus/Registry.hs

```diff
@@ -19,30 +20,28 @@ import qualified Control.Concurrent.STM as STM
 -- >>> :module +Prometheus
 -- >>> unregisterAll
 
-data RegisteredMetric = forall s. MkRegisteredMetric (Metric s)
-
-type Registry = [RegisteredMetric]
+type Registry = [IO [SampleGroup]]
```

Comment:
Please document this type.

Reply:
It's not actually exported, but I think it could still do with a note.

Reply:
For context, I think it's a bit unusual to have `IO` neither at the outermost level, nor at the bottom of a monad transformer stack—perfectly valid of course, just not something you see every day. Noting what we're doing & why will make it a little easier for future maintainers.